src/bert_layers/attention.py

```diff
@@ -330,11 +344,19 @@ def forward(
             attn = attn.view(bs, dim)
         else:
             qkv = bert_padding.pad_input(qkv, indices, cu_seqlens.shape[0] - 1, max_seqlen)  # batch, max_seqlen, thd
-            unpad_bs, *_ = qkv.shape
+            unpad_bs, seqlen, _ = qkv.shape
 
             qkv = qkv.view(unpad_bs, -1, 3, self.num_attention_heads, self.attn_head_size)
             q, k, v = qkv.transpose(3, 1).unbind(dim=2)  # b h s d
-            attn = F.scaled_dot_product_attention(q, k, v, dropout_p=self.p_dropout)
+            attn = F.scaled_dot_product_attention(
+                q,
+                k,
+                v,
+                dropout_p=self.p_dropout,
+                attn_mask=attn_mask[:, None, None, :seqlen].to(torch.bool).expand(unpad_bs, 1, seqlen, seqlen)
```

Comment:
Dumb question: in which case would the second dimension of attention mask be larger/different than seqlen?
Seems equals to me, but I might be missing something

Reply:
The Glue attention mask is larger. Didn't look into why.